### PR TITLE
[sc-133063] Automate circleci orb publishing

### PR DESCRIPTION
**Pull request by yusinto**
Note that the prerequisite of this work is having a github owner circleci token. There is a [security review](https://launchdarkly.atlassian.net/browse/SEC-773) for this. In the meantime to test this, we can a dev version of the orb.

This pr migrates existing Circleci shell commands to built-in releaser steps:

1. (Done in past pr) Modify`update-version.sh` to update version numbers in circleci/orb.yml.
2. Modify `publish.sh` to install the circleci cli
3. Modify `publish.sh` to run circleci setup authenticating with a circleci token that has the github owner role.
4. Modify `publish.sh` to run circleci orb publish

### CODE REVIEW

File: .ldrelease/publish-circleci.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+# Run this in publish step after all version information have been updated.
+set -ev
+
+brew install circleci
```
Comment (InTheCloudDan) [inline .ldrelease/publish-circleci.sh:6]: I don't think this should be in the script. We can run a `which` to verify the user has it installed and error out if it isn't there

File: .ldrelease/publish-circleci.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# Read 2 arguments from the command line so we can debug this script.
+# Argument 1 is the release version. Defaults to releaser env variable.
+RELEASE_VERSION=${1:-$LD_RELEASE_VERSION}
+
+# Argument 2 is the circleci token. Defaults releaser circleci secret.
+CIRCLECI_CLI_TOKEN=${2:-"${LD_RELEASE_SECRETS_DIR}/circleci_token"}
+CIRCLECI_CLI_HOST="https://circleci.com"
+
+# Validate circleci orb config. No publishing done on this step.
+circleci orb validate build/package/circleci/orb.yml || (echo "Unable to validate orb"; exit 1)
```
Comment (InTheCloudDan) [inline .ldrelease/publish-circleci.sh:17]: We have a `make` version of this do we want to call that so if there's changes required we only need to do them in one spot.

File: .ldrelease/publish-circleci.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+# dev publish only, not production
+circleci orb publish build/package/circleci/orb.yml launchdarkly/ld-find-code-refs@dev:$RELEASE_VERSION --token $CIRCLECI_CLI_TOKEN --host $CIRCLECI_CLI_HOST
```
Comment (InTheCloudDan) [inline .ldrelease/publish-circleci.sh:20]: Same as above, this should call the `make` command.

**Review by InTheCloudDan — CHANGES_REQUESTED** [review]

File: .ldrelease/publish-circleci.sh (hunk 1 above)
Comment (yusinto) [reply .ldrelease/publish-circleci.sh:6]: We need the `circleci` cli installed in the releaser environment. If we don't install it, we won't be able to automate any of the circleci steps. I could be missing something obvious here, but I don't think installing the circleci cli is an issue?

Comment (InTheCloudDan) [reply .ldrelease/publish-circleci.sh:6]: we should install it in the container if it's not there currently, `brew` is pretty mac dependent, while the releaser containers are using linux.

Comment (InTheCloudDan) [reply .ldrelease/publish-circleci.sh:6]: looks like it's based on Ubuntu. So we can change it use `apt-get` and I think it should be good to go. Thinking on it more yeah this script is really only used in the release process.

Comment (InTheCloudDan) [reply .ldrelease/publish-circleci.sh:6]: if we go this route though I think we'll need to install with curl to bash which isn't ideal https://circleci.com/docs/2.0/local-cli/#alternative-installation-method

Comment (yusinto) [reply .ldrelease/publish-circleci.sh]: Good point re `brew` being osx specific. I've changed this now to use apt to install curl and then installing circleci using curl.

File: .ldrelease/publish-circleci.sh (hunk 2 above)
Comment (yusinto) [reply .ldrelease/publish-circleci.sh:18]: I think we should remove those circleci make targets and use this shell script to publish to circle. I've added a comment to do this when we have a github owner token but will leave them for now so we can continue publishing the old way. In the future, all publishing to circleci orbs should be done through `publish-circleci.sh`.

**Review by InTheCloudDan — APPROVED** [review]

**Pull request merged**